service/src/main/java/org/cbioportal/service/impl/GenesetDataServiceImpl.java

```diff
@@ -92,6 +92,8 @@
                     GenesetMolecularData genesetData = new GenesetMolecularData();
                     genesetData.setMolecularProfileId(molecularProfileId);
                     genesetData.setSampleId(sample.getStableId());
+                    genesetData.setPatientId(sample.getPatientStableId());
```

Comment:
for optimal compatibility we also need uniqueSampleKey and uniquePatientKey like in other endpoints

Reply:
they already exist in this endpoint, it was just not working properly because patient and study id was not being set.

Reply:
oh okay